### ISSUE TITLE
User properties shows incorrect setting for "Grant option"

### ISSUE THREAD

**Issue by JTMosaic** [dbeaver/dbeaver#22351]
### Description

When a user is granted "ALL" access to all schemas in MySQL or MariaDB they are not granted the right to grant access to other users. This is called the "grant option". This appears incorrectly in user properties pages.

### DBeaver Version

23.3.0+

### Operating System

macOS 14.2.1

### Database and driver

MySQL 8
MariaDB 11

### Steps to reproduce

Granting a user permissions like so:

```
grant all on *.* to user;
```

Will create a grant entry like:

```show grants for user```
![image](https://github.com/dbeaver/dbeaver/assets/9417720/a7322b37-41ea-4589-b9ab-fd6676b72c0a)

Granting a user permissions like so:

```
grant all on *.* to user with grant option;
```

Will create a grant entry like:
```show grants for user```
![image](https://github.com/dbeaver/dbeaver/assets/9417720/85dc1e62-a98f-46ba-ba3a-febeb755ed17)

In both cases, the "Grant option" box is checked on the user properties page

![image](https://github.com/dbeaver/dbeaver/assets/9417720/a4006c35-2135-475a-a94f-b1cc0f78a24f)

**Comment by LonwoLonwo** [comment]
Thanks for the report @JTMosaic

**Comment by LonwoLonwo** [comment]
Fixed. The fix will be available in the next DBeaver version, 23.3.4, on 05-02-2024.
